pkg/manifest/os.go

```diff
@@ -881,6 +881,29 @@ func prependKernelCmdlineStage(pipeline osbuild.Pipeline, kernelOptions []string
 		)
 	}
 
+	if mountUnits {
+		// The systemd-remount-fs service reads /etc/fstab to discover mount
+		// options for / and /usr. Without an /etc/fstab, / and /usr do not get
+		// remounted, which means if they are mounted read-only in the initrd,
+		// they will remain read-only. Flip the option if we're using only
+		// mount units, otherwise the filesystems will stay mounted 'ro'.
+		//
+		// See https://www.freedesktop.org/software/systemd/man/latest/systemd-remount-fs.service.html
+		for idx := range kernelOptions {
+			if kernelOptions[idx] == "ro" {
+				kernelOptions[idx] = "rw"
+				break
+			}
```

Comment:
It seems like it'd be cleaner to ensure that `rw` is injected from the start if mount units are in use.

Reply:
You're absolutely correct.  I don't like switching options this late in the manifest generation process either.  Making the change earlier in the process might need more refactoring though, so if it turns out to be too much I might leave it for a later cleanup.

Reply:
I'm going to leave this for another time, because `ro` is a very common option we statically add to a lot of image definitions.  Making it conditional on the mount units option from the top would need a lot of changes that would complicate (and perhaps delay) this PR.

Reply:
I understand, it makes sense.

However, I would also say that I think the `ro` option doesn't make sense to use anymore *anywhere* (see also e.g. https://github.com/ostreedev/ostree/issues/3193#issuecomment-2578237486 )

Reply:
ah!  That's great to know.  Happy to drop it everywhere, though might have to do it carefully for RHEL in case there are any side effects that users rely on.